core/src/main/java/org/geysermc/geyser/command/defaults/ConnectionTestCommand.java

```diff
@@ -102,37 +142,67 @@ public void execute(@Nullable GeyserSession session, GeyserCommandSource sender,
                             "See here for steps on how to resolve: " + "https://wiki.geysermc.org/geyser/fixing-unable-to-connect-to-world/#using-geyser-on-the-same-computer");
                 }
 
-                // mcsrvstatus will likely be replaced in the future with our own service where we can also test
-                // around the OVH workaround without worrying about caching
-                JsonNode output = WebUtils.getJson("https://api.mcsrvstat.us/bedrock/2/" + args[0]);
+                // Generate some random, unique bits that another server wouldn't provide
+                byte[] random = new byte[2];
+                new Random().nextBytes(random);
+                StringBuilder randomStr = new StringBuilder();
+                for (byte b : random) {
+                    randomStr.append(Integer.toHexString(b));
+                }
+                String connectionTestMotd = "Geyser Connection Test " + randomStr;
+                CONNECTION_TEST_MOTD = connectionTestMotd;
+
+                sender.sendMessage("Testing server connection now. Please wait...");
+                JsonNode output;
+                try {
+                    output = WebUtils.getJson("https://checker.geysermc.org/ping?hostname=" + ip + "&port=" + port);
+                } finally {
+                    CONNECTION_TEST_MOTD = null;
+                }
 
-                long cacheTime = output.get("debug").get("cachetime").asLong();
+                JsonNode cache = output.get("cache");
                 String when;
-                if (cacheTime == 0) {
-                    when = "now";
+                if (cache.get("fromCache").asBoolean()) {
+                    when = cache.get("secondsSince").asInt() + " seconds ago";
                 } else {
-                    when = ((System.currentTimeMillis() / 1000L) - cacheTime) + " seconds ago";
+                    when = "now";
                 }
 
-                if (output.get("online").asBoolean()) {
-                    sender.sendMessage("Your server is likely online as of " + when + "!");
+                if (output.get("success").asBoolean()) {
+                    JsonNode ping = output.get("ping");
+                    JsonNode pong = ping.get("pong");
+                    String remoteMotd = pong.get("motd").asText();
+                    if (!connectionTestMotd.equals(remoteMotd)) {
+                        sender.sendMessage("The MOTD did not match when we pinged the server (we got '" + remoteMotd + "'). " +
+                                "Did you supply the correct IP and port of your server?");
+                        sendLinks(sender);
+                        return;
+                    }
+
+                    if (ping.get("tcpFirst").asBoolean()) {
+                        sender.sendMessage("Your server hardware likely has some sort of firewall preventing people from joining easily. See LINK for more information.");
```

Comment:
LINK?

Reply:
Ill replace it with a proper link once https://github.com/GeyserMC/GeyserWiki/pull/194 is merged, the OVH stuff will then be on a separate portforwarding page

Reply:
Now changed to the final quicklink https://geysermc.link/ovh-firewall - thanks to said PR being merged & due to Chew being awesome with the quicklink :p